File: btap-modeling/lib/btap_modeling/systems/baseboards_only.rb
module BtapModeling
  module Systems
    # Zone baseboards with no air system (CBECS 'Baseboard electric' / 'Baseboard gas
    # boiler'): electric convective baseboards, or hot-water baseboards served by the
    # (gas) boiler loop. Heating-only; builds no air loops.
    class BaseboardsOnly < BaseSystem
      # @return [Array<OpenStudio::Model::AirLoopHVAC>] empty (no air system)
      def build(model, zones, control_zone: nil, namer: :default, hw_loop: nil, chw_loop: nil)
        baseboard_type = config['baseboard_type']
        zones.each do |zone|
          Baseboards.add(model, zone, baseboard_type: baseboard_type, hw_loop: hw_loop)
        end
        []
      end
    end
  end
end

File: python/btap/modeling/hvac/systems/baseboards_only.py
"""Zone baseboards with no air system (CBECS 'Baseboard electric' / 'Baseboard gas
boiler'): electric convective baseboards, or hot-water baseboards served by the
(gas) boiler loop. Heating-only; builds no air loops."""

from __future__ import annotations

from btap.modeling.hvac.systems import baseboards
from btap.modeling.hvac.systems.base_system import BaseSystem


class BaseboardsOnly(BaseSystem):
    def build(self, model, zones, control_zone=None, namer='default',
              hw_loop=None, chw_loop=None):
        """:return: list of openstudio.model.AirLoopHVAC — empty (no air system)"""
        baseboard_type = self.config['baseboard_type']
        for zone in zones:
            baseboards.add(model, zone, baseboard_type, hw_loop=hw_loop)
        return []
